test-crawler/extractor/extractor.go

```diff
@@ -204,17 +232,41 @@ func getFunctionNodes(content string, treeCursor *sitter.TreeCursor, parser *a.P
 				}
 
 				funcAnnoPair = append(funcAnnoPair, struct {
-					Node *sitter.Node
-					Name string
+					Node     *sitter.Node
+					Function c.FunctionAnnotation
 				}{
 					Node: node,
-					Name: funcName,
+					Function: c.FunctionAnnotation{
+						Name: funcName,
+					},
 				})
 
 				node = nil
 				funcName = ""
 				isIgnored = false
 			}
+			if child.Type() == string(METHOD_DECLARATION) {
+				funcName = content[child.Child(2).StartByte():child.Child(2).EndByte()]
```

Comment:
Add comment(s) explaining what the Child(2,3,4...) are.

Reply:
A bit too long, but good. I would remove this:
```
First iteration through children nodes:
Child.Type = method_declaration
Child.Value = whole function body func(*x)(a,b)c{...}
```
Since we are not handling the type/value in this part of the function. Also, I would remove the "has a type XX, and represents..." text, because it is superfluous.